packages/neo-one-client/src/__e2e__/DeveloperClient.test.js

```diff
@@ -143,67 +167,110 @@ async function confirmTransaction(
   test('runConsensusNow', async () => {
-    const network = 'e2e-1';
+    const networkName = 'e2e-1';
     const walletName = 'wallet-1';
 
-    const { developerClient, client, master, wallet } = await setupClients(
-      network,
-      walletName,
+    const { developerClient, client, keystore, master } = await setupClients(
+      networkName,
     );
-    const transaction = await setupTransaction(client, master, wallet);
+    const wallet = await addWallet({ walletName, keystore, networkName });
+
+    const transaction = await setupTransaction({ client, master, wallet });
 
-    await confirmTransaction(transaction, developerClient);
+    await Promise.all([
+      confirmTransaction(transaction),
+      developerClient.runConsensusNow(),
+    ]);
 
-    await checkWalletBalance(walletName, network);
+    await checkWalletBalance({ walletName, networkName });
   });
 
   test('updateSettings', async () => {
-    const network = 'e2e-2';
+    const networkName = 'e2e-2';
     const walletName = 'wallet-2';
 
-    const { developerClient, client, master, wallet } = await setupClients(
-      network,
-      walletName,
+    const { developerClient, client, keystore, master } = await setupClients(
+      networkName,
     );
+    const wallet = await addWallet({ walletName, keystore, networkName });
 
     await developerClient.updateSettings({ secondsPerBlock: 1 });
     await new Promise(resolve => setTimeout(() => resolve(), 15000));
-    const transaction = await setupTransaction(client, master, wallet);
+    const transaction = await setupTransaction({ client, master, wallet });
 
     await confirmTransaction(transaction);
 
-    await checkWalletBalance(walletName, network);
+    await checkWalletBalance({ walletName, networkName });
   });
 
   test('fastForwardOffset', async () => {
-    const network = 'e2e-3';
+    const networkName = 'e2e-3';
+    const secondsPerBlock = 1;
     const offsetSeconds = 1000000000;
 
-    const { developerClient, client } = await setupClients(network);
-    await developerClient.fastForwardOffset(offsetSeconds);
-    await new Promise(resolve => setTimeout(() => resolve(), 15000));
+    const { developerClient, client } = await setupClients(networkName);
 
-    const firstBlock = await client.read(network).getBlock(0);
-    const secondBlock = await client.read(network).getBlock(1);
+    await developerClient.updateSettings({ secondsPerBlock });
+    await new Promise(resolve => setTimeout(() => resolve(), 15000));
 
-    expect(secondBlock.time - firstBlock.time).toBeGreaterThanOrEqual(
-      offsetSeconds,
-    );
+    await developerClient.fastForwardOffset(offsetSeconds);
+    await new Promise(resolve => setTimeout(() => resolve(), 2000));
+
+    const times = await getBlockTimes({ client, networkName });
+
+    let offsetFound = false;
+    for (let i = 0; i < times.length - 2; i += 1) {
+      const offset = times[i + 1] - times[i];
+      if (offset === offsetSeconds + secondsPerBlock) {
```

Comment:
This seems kinda janky and possibly flaky. I'd just check for `offset >= offsetSeconds` and not rely on `secondsPerBlock` at all.